src/metadata.h

```diff
@@ -0,0 +1,58 @@
+/*
+Minetest
+Copyright (C) 2010-2013 celeron55, Perttu Ahola <celeron55@gmail.com>
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation; either version 2.1 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along
+with this program; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+#ifndef METADATA_HEADER
+#define METADATA_HEADER
+
+#include "irr_v3d.h"
+#include <iostream>
+#include <vector>
+#include "util/string.h"
+
+class Metadata
+{
+public:
+	void clear();
+	bool empty() const;
+	size_t size() const;
+	bool contains(const std::string &name) const;
+
+	std::string serialize() const;
+	void deSerialize(const std::string &in);
+
+	bool operator==(const Metadata &other) const;
+	inline bool operator!=(const Metadata &other) const
+	{
+		return !(*this == other);
+	}
+
+	// Generic key/value store
+	std::string getString(const std::string &name, unsigned short recursion = 0) const;
+	void setString(const std::string &name, const std::string &var);
+	// Support variable names in values
+	std::string resolveString(const std::string &str, unsigned short recursion = 0) const;
+	StringMap getStrings() const
+	{
+		return m_stringvars;
+	}
+protected:
+	StringMap m_stringvars;
```

Comment:
StringMap is an ordered map, please use UNORDERED_MAP<std::string, std::string> instead. ordering is not needed there